### ISSUE TITLE
Springboot should check diskspace before booting up

### ISSUE THREAD

**Issue by ravisachar** [spring-projects/spring-boot#13407]
Operating System: Centos 7.0
Architecture: 64 bit
Java Version:  "1.8.0_112"
Springboot Version: 1.5.1.RELEASE

Scenario: When the OS disk-space is full say 100%, the springboot app throws the below exception while starting up.

```
2018-06-07 06:11:11 [main] INFO  o.s.d.r.c.RepositoryConfigurationDelegate - Multiple Spring Data modules found, entering strict repository configuration mode!
2018-06-07 06:11:12 [main] INFO  o.s.b.f.a.AutowiredAnnotationBeanPostProcessor - JSR-330 'javax.inject.Inject' annotation found and supported for autowiring
2018-06-07 06:11:12 [main] INFO  o.s.c.s.PostProcessorRegistrationDelegate$BeanPostProcessorChecker - Bean 'org.springframework.boot.autoconfigure.validation.ValidationAutoConfiguration' of type [class org.springframework.boot.autoconfigure.validation.ValidationAutoConfiguration] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2018-06-07 06:11:12 [main] INFO  o.s.c.s.PostProcessorRegistrationDelegate$BeanPostProcessorChecker - Bean 'validator' of type [class org.springframework.validation.beanvalidation.LocalValidatorFactoryBean] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2018-06-07 06:11:12 [main] INFO  o.s.c.s.PostProcessorRegistrationDelegate$BeanPostProcessorChecker - Bean 'org.springframework.transaction.annotation.ProxyTransactionManagementConfiguration' of type [class org.springframework.transaction.annotation.ProxyTransactionManagementConfiguration$$EnhancerBySpringCGLIB$$b5adc3a4] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2018-06-07 06:11:13 [main] INFO  o.s.b.c.e.t.TomcatEmbeddedServletContainer - Tomcat initialized with port(s): 8095 (http)
2018-06-07 06:11:13 [main] INFO  o.a.catalina.core.StandardService - Starting service Tomcat
2018-06-07 06:11:13 [main] INFO  o.a.catalina.core.StandardEngine - Starting Servlet Engine: Apache Tomcat/8.5.13
2018-06-07 06:11:13 [Tomcat-startStop-1] ERROR o.a.catalina.core.ContainerBase - A child container failed during start
```

This exception is misleading, Could springboot app check the disk-space before starting up and display the below message in case if there is not enough storage
e.g. "No enough disk space ..exiting" 

Impact: This would save lot of time for troubleshooting the root-cause

**Comment by bclozel** [comment]
I don't think this is the application should be responsible for that - this is a concern for the deployment solution. Also, given the wide range of deployment solutions, it's impossible to consistently check for such issues. Should we check the available space? The usable space? On which partition(s)? What about write permissions? Should we check firewall rules to make sure the app can be reached?

I'm closing this issue as in my opinion it is out of scope for Spring Boot.

Thanks,
